apps/api/src/routes/competitions.routes.ts

```diff
@@ -454,5 +454,181 @@ export function configureCompetitionsRoutes(
+   *                       description: Competition name
+   *                     description:
+   *                       type: string
+   *                       nullable: true
+   *                       description: Competition description
+   *                     externalLink:
+   *                       type: string
+   *                       nullable: true
+   *                       description: External URL for competition details
+   *                     imageUrl:
+   *                       type: string
+   *                       nullable: true
+   *                       description: URL to competition image
+   *                     status:
+   *                       type: string
+   *                       enum: [pending, active, completed]
+   *                       description: Competition status
+   *                     crossChainTradingType:
+   *                       type: string
+   *                       enum: [disallowAll, disallowXParent, allow]
+   *                       description: The type of cross-chain trading allowed in this competition
+   *                     startDate:
+   *                       type: string
+   *                       format: date-time
+   *                       nullable: true
+   *                       description: Competition start date (null for pending competitions)
+   *                     endDate:
+   *                       type: string
+   *                       format: date-time
+   *                       nullable: true
+   *                       description: Competition end date (null for pending/active competitions)
+   *                     createdAt:
+   *                       type: string
+   *                       format: date-time
+   *                       description: When the competition was created
+   *                     updatedAt:
+   *                       type: string
+   *                       format: date-time
+   *                       description: When the competition was last updated
+   *       400:
+   *         description: Bad request - Invalid competition ID format
+   *       404:
+   *         description: Competition not found
+   *       401:
+   *         description: Unauthorized - Missing or invalid authentication
+   *       500:
+   *         description: Server error
+   */
+  router.get("/:competitionId", controller.getCompetitionById);
+
+  /**
+   * @openapi
+   * /api/competitions/{competitionId}/agents:
+   *   get:
+   *     tags:
+   *       - Competition
+   *     summary: Get agents participating in a competition
+   *     description: Get a list of all agents participating in a specific competition with their scores and positions
+   *     security:
+   *       - BearerAuth: []
+   *     parameters:
+   *       - in: path
+   *         name: competitionId
+   *         schema:
+   *           type: string
+   *         required: true
+   *         description: The ID of the competition to get agents for
```

Comment:
Based on the api spec google doc we need filter, sort, limit, and offset querystring params for this endpoint